### DISCUSSION TITLE
Azure SSO Issues

### DISCUSSION THREAD

**Discussion by avnetworking** [netbox-community/netbox#17845] [Help Wanted!]
Hello!

I have been following [this guide ](https://netboxlabs.com/docs/netbox/en/stable/administration/authentication/microsoft-azure-ad/) and am able to get auth with Azure, but get the error: 

![image](https://github.com/user-attachments/assets/8fec22fe-7b7c-4900-8485-0b0daa3e4f38)

I have done some searching and everyone seems to end with a minor typing error, however i have rekeyed everything, even blown the image away and started from scratch. I end up at the same state. Any help on where to look? Is there any configuration in Netbox required outside what is done in the configuration.py? 

As a note I have tried creating myself a matching local account, adding a default group that matches what is in Netbox, and a whole bunch of other tweaks like SOCIAL_AUTH_PIPELINE with no changes. The biggest change I got was using REMOTE_AUTH_AZUREAD_OATH2_RESOURCE= 'example' with this in place I would get an error pointing to that specific resource was not found.

Any help would be greatly appreciated as I am up a creek without a paddle at this point.

**Comment by rc-networks** [comment]
@avnetworking 
I'm facing the same issue after upgrading to v4.0

Did you managed to find any solution?

**Comment by avnetworking** [comment]
I moved on to Google SSO. I figured out the secret was wrong for Azure SSO, but then got an auth canceled event. Tried Google auth and it worked first shot. 

I would recommend rekeying the secret. That solved this issue, but I think something is broken somewhere. I even tried from a fresh install and ended at the same error. All my research brings up version issues, so possibly if your seeing this after v4 upgrade it could be some sort of version issue between Azure and the Social app plugin.
